src/MqttMailingService.cpp

```diff
@@ -87,17 +93,33 @@ __attribute__((unused)) void MqttMailingService::startWithDelegatedWiFi() {
     startWithDelegatedWiFi(WIFI_SSID_OVERRIDE, WIFI_PW_OVERRIDE, false);
 }
 
-__attribute__((unused)) void
+__attribute__((unused)) bool
 MqttMailingService::setBrokerURI(const char* brokerURI) {
-    if (strlen(brokerURI) > 64) {
+    if (strlen(brokerURI) > MAX_URI_LENGTH) {
         ESP_LOGW(TAG,
-                 "Warning: requested broker URI \"%s\" is too long: %i "
-                 "characters (max %i). It got truncated.",
-                 brokerURI, strlen(brokerURI), 63);
+                 "Error: requested broker URI \"%s\" is too long: %i "
+                 "characters (max %i).",
+                 brokerURI, strlen(brokerURI), MAX_URI_LENGTH);
+        return false;
     }
-    strncpy(mBrokerFullURI, brokerURI, 63);
+    strncpy(mBrokerFullURI, brokerURI, MAX_URI_LENGTH); 
+    return true;
 }
 
+__attribute__((unused)) bool setBroker(const char* brokerDomain, const bool hasSsl) {
+        char fullBrokerUri[MAX_URI_LENGTH + 1];
+        const char* protocol = hasSsl ? "mqtts" : "mqtt";
+        const int mqttPort = hasSsl ? 8883 : 1883;
+        if(sprintf(fullBrokerUri, "%s://%s:%d", protocol, brokerDomain, mqttPort) > 0) {
```

Comment:
this doesn't make sense. sprintf returns the number of characters written to the string, which is expected to be > 0.

Furthermore, this function does nothing. fullBrokerURI is a local vasriable and is thrown away after the function call.